src/components/cube-column-chooser.jsx

```diff
@@ -0,0 +1,76 @@
+import React, { useRef, useState } from 'react';
+import PropTypes from 'prop-types';
+import useClickOutside from './use/click-outside';
+import './cube-column-chooser.pcss';
+
+function findAttribute(event, attrName) {
+  let el = event.target;
+  while (!!el && !el.getAttribute(attrName)) {
+    el = el.parentElement;
+  }
+  const field = el && el.getAttribute(attrName);
+  return field;
+}
+
+export default function CubeColumnChooser({
+  alignTo, arrowStyle, chooseColumn, selectableColumns,
+}) {
+  const [filter, setFilter] = useState('');
+  const inputRef = useRef(null);
+  const selfRef = useRef(null);
+
+  useClickOutside(selfRef, true, () => {
+    chooseColumn(null);
+  });
+
+  const filteredColumnsOptions = selectableColumns.filter(item => item.title.toLowerCase().indexOf(filter) >= 0);
+
+
+  function updateFilter(e) {
+    const { value } = e.target;
+    setFilter(value.toLowerCase());
+  }
+
+  function selectRow(e) {
+    const title = findAttribute(e, 'data-title');
```

Comment:
Not too fond of us relying on the DOM to transport data, can we solve this differently?